src/renderer/atlas/BackendD2D.cpp

```diff
@@ -125,22 +126,43 @@ void BackendD2D::_handleSettingsUpdate(const RenderingPayload& p)
         _builtinGlyphsRenderTargetActive = false;
     }
 
-    if (renderTargetChanged || fontChanged || cellCountChanged)
+    if (renderTargetChanged || fontChanged || cellCountChanged || backgroundColorChanged)
     {
         const D2D1_BITMAP_PROPERTIES props{
             .pixelFormat = { DXGI_FORMAT_R8G8B8A8_UNORM, D2D1_ALPHA_MODE_PREMULTIPLIED },
             .dpiX = static_cast<f32>(p.s->font->dpi),
             .dpiY = static_cast<f32>(p.s->font->dpi),
         };
         const D2D1_SIZE_U size{
-            p.s->viewportCellCount.x,
-            p.s->viewportCellCount.y,
+            p.s->viewportCellCount.x + 2u,
+            p.s->viewportCellCount.y + 2u,
         };
         const D2D1_MATRIX_3X2_F transform{
             .m11 = static_cast<f32>(p.s->font->cellSize.x),
             .m22 = static_cast<f32>(p.s->font->cellSize.y),
+            .dx = -static_cast<f32>(p.s->font->cellSize.x),
+            .dy = -static_cast<f32>(p.s->font->cellSize.y),
         };
-        THROW_IF_FAILED(_renderTarget->CreateBitmap(size, nullptr, 0, &props, _backgroundBitmap.put()));
+
+        /*
+         We're allocating a bitmap that is one pixel wider on every side than the viewport so that we can fill in the gutter
+         with the background color. D2D doesn't have an equivalent to D3D11_TEXTURE_ADDRESS_BORDER, which we use in the D3D
+         backend to ensure the colors don't bleed off the edges.
+
+         XXXXXXXXXXXXXXXX <- background color
+         X+------------+X
+         X| viewport   |X
+         X|            |X
+         X|            |X
+         X+------------+X
+         XXXXXXXXXXXXXXXX
+
+         The translation in `transform` ensures that we render it off the top left of the render target.
+        */
+        auto backgroundFill = std::make_unique_for_overwrite<u32[]>(static_cast<size_t>(size.width * size.height));
+        std::fill_n(backgroundFill.get(), size.width * size.height, u32ColorPremultiply(p.s->misc->backgroundColor));

```

Comment:
A heap alloc for 1440 bytes? At least it only happens when we need to recreate the bitmap.

Reply:
Thinking about it some more, you could use `CreateCompatibleRenderTarget` and `Clear()` to initialize the bitmap. There's some code you could copy in `_resizeCursorBitmap`.

Another option is that we preallocate the foreground/background bitmap with a 1px margin and adjust the "stride" accordingly so that D3D is unaffected by this. The latter option is potentially the optimal solution, I think. (I'd be happy to help with that if you'd like.)

Reply:
Arguably more expensive than a heap alloc of 1440 bytes, FWIW.

A compatible render target, pointed to the bitmap, based on the current render target, all to just clear it (after converting the background color to D2D's preferred `D2D_COLOR_F`) and suck the bitmap out to stride-copy the actual background colors in.